actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1263,6 +1273,16 @@ def singleton?; false; end
         end
 
         class SingletonResource < Resource # :nodoc:
+          class << self
+            def default_actions(api_only)
+              if api_only
+                [:show, :create, :update, :destroy]
+              else
+                [:show, :create, :update, :destroy, :new, :edit]
+              end
```

Comment:
We can have a [similar approach](https://github.com/rails/rails/pull/51464/files#r1621482919) here.

Reply:
See reply on previous comment.